inc/calendar.class.php

```diff
@@ -196,8 +196,9 @@ static function processMassiveActionsForOneItemtype(MassiveAction $ma, CommonDBT
    /**
     * Clone a calendar to another entity : name is updated
     *
-    * @param $options array of new values to set
-    * @return boolean True on success
+    * @param $options array Array of new values to set
+    * @return boolean|integer The new ID on success or false on failure
```

Comment:
I guess the original `duplicate` method should still return only a boolean. Since this is deprecated, this sounds a bad idea to change what it returns, since we should need to handle usage a bit differently.

Reply:
I think my thought on that was any code checking for just a true/false return should still work if an ID is returned. Any added item should have an ID >= 1 and be resolved as a true value indicating a success. There isn't any instance in the core or any plugin I have installed, but it is possible something uses a strict comparison of `$calendar->duplicate() === true)`.

Reply:
"but it is possible this is an issue if something uses a strict comparison" => yes, that was what I had in mind; and the change is not mandatory ;)

Reply:
And also, you have to add a `Toolbox::deprecated()` call at the beginning of the method